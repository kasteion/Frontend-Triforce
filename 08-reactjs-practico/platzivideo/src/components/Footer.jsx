import React from 'react'
import '../assets/styles/components/Footer.scss'

const Footer = () => {
    return (
        <footer className="footer">
            <a className="footer--anchor" href="#">Terminos de uso</a>
            <a className="footer--anchor" href="#">Declaraci√≥n de privacidad</a>
            <a className="footer--anchor" href="#">Centro de ayuda</a>
        </footer>
    )
}

export default Footer