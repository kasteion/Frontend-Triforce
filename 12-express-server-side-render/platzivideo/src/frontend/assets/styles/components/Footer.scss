@import '../Vars.scss';

.footer {
    padding: 30px;
}

.footer--anchor {
    color: white;
    text-decoration: none;
    margin: 5px 15px 5px 15px;
}

.footer--anchor:hover {
    text-decoration: underline;
}