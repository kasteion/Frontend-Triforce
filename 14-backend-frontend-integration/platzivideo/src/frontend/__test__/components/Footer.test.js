import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
//import Adapter from 'enzyme-adapter-react-16'
//Importamos el componente que vamos a probar...
import Footer from '../../components/Footer';

//configure({ adapter: new Adapter()});
//Luego necesitamos crear una suite (una suite de pruebas)
//describe('', ()=> {})
describe('<Footer />', () => {
  const footer = mount(<Footer />);
  //Lo primero que quiero probar es que exista el componente
  //(que exista un componente que haga render)
  test('Render Footer Component', () => {
    expect(footer.length).toEqual(1);
  });

  test('Footer haves 3 anchors', () => {
    expect(footer.find('a')).toHaveLength(3);
  });

  // Prueba con snapshot
  test('Footer Snapshot', () => {
    const footer = create(<Footer />);
    expect(footer.toJSON()).toMatchSnapshot();
  });
});
