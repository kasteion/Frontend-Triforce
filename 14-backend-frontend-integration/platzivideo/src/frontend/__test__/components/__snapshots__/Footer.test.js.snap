// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Footer Snapshot 1`] = `
<footer
  className="footer"
>
  <a
    className="footer--anchor"
    href="/"
  >
    Terminos de uso
  </a>
  <a
    className="footer--anchor"
    href="/"
  >
    Declaraci√≥n de privacidad
  </a>
  <a
    className="footer--anchor"
    href="/"
  >
    Centro de ayuda
  </a>
</footer>
`;
